### PR TITLE
[New Rule] Lolbas ImageLoad via Windows Update Client

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/365

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+process where event.type in ("start", "process_started") and
+  (process.pe.original_file_name == "wuauclt.exe" or process.name == "wuauclt.exe") and
+  
+   /* necessary windows update client args to load a dll*/
+   
+   wildcard(process.args, "/RunHandlerComServer") and wildcard(process.args, "/UpdateDeploymentProvider") and
+   
+   /* common paths writeable by a standard user where the target DLL can be placed */
+   
+   wildcard(process.args, "C:\\Users\\*", "C:\\ProgramData\\*", "C:\\Windows\\Temp\\*", "C:\\Windows\\Tasks\\*")
```
Comment (brokensound77) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:34]:
```suggestion
   wildcard(process.args, "C:\\Users\\*.dll", "C:\\ProgramData\\*.dll", "C:\\Windows\\Temp\\*.dll", "C:\\Windows\\Tasks\\*.dll")
```

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+rule_id = "edf8ee23-5ea7-4123-ba19-56b41e424ae3"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timeline_id = "76e52245-7519-4251-91ab-262fb1a1728c"
+type = "eql"
+
+query = '''
+
+process where event.type in ("start", "process_started") and
+  (process.pe.original_file_name == "wuauclt.exe" or process.name == "wuauclt.exe") and
```
Comment (brokensound77) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:26]: pending `process.pe.original_file_name` being mapped into `winlogbeat`

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  
+   /* necessary windows update client args to load a dll*/
```
Comment (brokensound77) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:28]:
```suggestion
   /* necessary windows update client args to load a dll */
```

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2020/10/13"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/13"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies Execution via abusing the Windows Update Client to load an arbitrary Image."
```
Comment (brokensound77) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:9]:
```suggestion
description = "Identifies Execution via abusing the Windows Update Client to load an arbitrary DLL."
```

Comment (brokensound77) [reply rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:9]: I would also add a line about how this is used to evade defenses. I am sure all LOLbins have some similar explanatory verbiage.

**Review by brokensound77 — COMMENTED** [review]
Nice rule, I left some comments. Was the primary reason for EQL here to handle case in `process.args`?

**Comment by brokensound77** [comment]
Blocked until `process.pe.original_file_name` is mapped into winlogbeat

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml (hunk 2 above)
Comment (Samirbous) [reply rules/windows/defense_evasion_execution_lolbas_wuauclt.toml]: removed blocked for process.pe.ofn

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+[metadata]
+creation_date = "2020/10/13"
+maturity = "production"
+updated_date = "2020/10/13"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies Execution via abusing the Windows Update Client to load an arbitrary DLL."
```
Comment (bm11100) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:8]:
```suggestion
description = "Identifies abuse of the Windows Update Auto Update Client (wuauclt.exe) to load an arbitrary DLL."
```

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+from = "now-9m"
+index = ["logs-endpoint.events.*", "winlogbeat-*"]
+language = "eql"
+license = "Elastic License"
+name = "ImageLoad via Windows Update Client"
```
Comment (bm11100) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:13]:
```suggestion
name = "ImageLoad via Windows Update Auto Update Client"
```

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+references = ["https://dtm.uk/wuauclt/"]
+risk_score = 47
+rule_id = "edf8ee23-5ea7-4123-ba19-56b41e424ae3"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+type = "eql"
+
+query = '''
+
```
Comment (bm11100) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:22]: we don't need all the blank lines 

```
process where event.type in ("start", "process_started") and
  (process.pe.original_file_name == "wuauclt.exe" or process.name : "wuauclt.exe") and
  
   /* necessary windows update client args to load a dll */
   process.args : "/RunHandlerComServer" and process.args : "/UpdateDeploymentProvider" and
   
   /* common paths writeable by a standard user where the target DLL can be placed */
   process.args : ("C:\\Users\\*.dll", "C:\\ProgramData\\*.dll", "C:\\Windows\\Temp\\*.dll", "C:\\Windows\\Tasks\\*.dll")
```

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+name = "ImageLoad via Windows Update Auto Update Client"
+references = ["https://dtm.uk/wuauclt/"]
+risk_score = 47
+rule_id = "edf8ee23-5ea7-4123-ba19-56b41e424ae3"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+type = "eql"
+
+query = '''
+
```
Comment (dstepanic17) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:22]:
```suggestion
```

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+process where event.type in ("start", "process_started") and
+  (process.pe.original_file_name == "wuauclt.exe" or process.name : "wuauclt.exe") and
+   /* necessary windows update client args to load a dll */
+   process.args : "/RunHandlerComServer" and process.args : "/UpdateDeploymentProvider" and
+   /* common paths writeable by a standard user where the target DLL can be placed */
+   process.args : ("C:\\Users\\*.dll", "C:\\ProgramData\\*.dll", "C:\\Windows\\Temp\\*.dll", "C:\\Windows\\Tasks\\*.dll")
+
```
Comment (dstepanic17) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:29]:
```suggestion
```

**Review by dstepanic17 — COMMENTED** [review]
LGTM, I just suggested a few changes to be consistent with the other rules and remove line breaks in query section.

File: rules/windows/defense_evasion_execution_lolbas_wuauclt.toml [hunk 10]
```diff
@@ -0,0 +1,42 @@
+[metadata]
+creation_date = "2020/10/13"
+maturity = "production"
+updated_date = "2020/10/13"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies abuse of the Windows Update Auto Update Client (wuauclt.exe) to load an arbitrary DLL."
```
Comment (bm11100) [inline rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:8]:
```suggestion
description = "Identifies abuse of the Windows Update Auto Update Client (wuauclt.exe) to load an arbitrary DLL. This behavior is used as a defense evasion technique to blend-in malicious activity with legitimate Windows software."
```

Comment (bm11100) [reply rules/windows/defense_evasion_execution_lolbas_wuauclt.toml:8]: re lint

**Review by bm11100 — APPROVED** [review]
lgtm

**Review by dstepanic17 — APPROVED** [review]

**Pull request merged**